File: frontend/campus_app/lib/models/story_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'story_dto.g.dart';

@JsonSerializable()
class StoryDTO {
  final String storyId; // UUID'ler String olarak temsil edilir
  final String profilePhoto; // Profil fotoğrafı
  final String username; // Kullanıcı adı
  final int userId; // Kullanıcı ID'si
  final String photo; // Hikaye fotoğrafı
  final int score; // Hikaye skoru

  StoryDTO({
    required this.storyId,
    required this.profilePhoto,
    required this.username,
    required this.userId,
    required this.photo,
    required this.score,
  });

  factory StoryDTO.fromJson(Map<String, dynamic> json) => _$StoryDTOFromJson(json);
  Map<String, dynamic> toJson() => _$StoryDTOToJson(this);
} 

File: backend/src/main/java/bingol/campus/story/core/response/StoryDTO.java
package bingol.campus.story.core.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class StoryDTO {
    private UUID storyId;
    private String profilePhoto;
    private String username;
    private long userId;
    private String photo;
    private int score;
}
